src/rade_api.h

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------*\
+
+  rade_api.h
+
+  Library of API functions that implement the Radio Autoencoder API.
+
+\*---------------------------------------------------------------------------*/
+
+/*
+  Copyright (C) 2024 David Rowe
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions
+  are met:
+
+  - Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  - Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR
+  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef __RADE_API__
+#define __RADE_API__
+
+#include <sys/types.h>
+
+// This declares a single-precision (float) complex number
+
+#ifndef __RADE_COMP__
+#define __RADE_COMP__
+typedef struct {
+  float real;
+  float imag;
+} RADE_COMP;
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Sample rates used
+#define RADE_FS_8000 8000           // modem waveform sample rate
+#define RADE_FS_16000 16000         // speech sample rate
+
+// note single context only in this version, one context has one Tx, and one Rx
+struct rade *rade_open(char model_file[]);
+void rade_close(struct rade *r);
+
+// Allows API users to determine if the API has changed
+int rade_version(void);
+
+// helpers to set up arrays
+int rade_n_tx_out(struct rade *r);
```

Comment:
Speaking of documentation, maybe a comment explaining what each of these are. It took me a minute to realize that this line is for the number of TX output samples, for example.

Reply:
Well a minute is pretty good, and the demo programs should help.  Would a name change help? I was trying to make the function names self-explanatory. But feel free to add extra comments/doc if you like.

Reply:
It might help, but then again, adding Doxygen documentation could also help (and is already slated for v2).